src/routes/identity/account/+page.svelte

```diff
@@ -29,26 +30,12 @@
 		copied = true
 	}
 
-	async function getBalances() {
+	onMount(() => {
 		const address = $walletStore.wallet?.address
 		if (address === undefined) return
 
-		const balance = await getBalance(address)
-
-		balanceStore.update((balanceState) => ({
-			...balanceState,
-			balances: balanceState.balances.map((value) => {
-				if (value.address) {
-					return value
-				} else {
-					return {
-						...value,
-						amount: balance,
-					}
-				}
-			}),
-		}))
-	}
+		adapter.initializeBalances(address)
+	})
```

Comment:
In the `firebase` adapter I do this when the user logs in and the balances are fetched. I wonder if that would also work. This way, we don't need to go to `/identity/account` to initialize the balances. It happens after creating/restoring wallet.

https://github.com/logos-innovation-lab/waku-objects-playground/blob/main/src/lib/adapters/firebase/index.ts#L193

Reply:
Oh I see this is also resolved no and is in `onLogIn`

Reply:
It's still there and also in the `waku` adapter. However if the balance is only checked once the user logs in then the account page may show an outdated balance when you see it. I had this exact problem during testing and that's why I added the button there, but the `onMount` version is better.